Scripts/Runtime/Core/CollectionsRegistry.cs

```diff
@@ -323,15 +303,50 @@ public void PrepareForEditorMode()
             for (int i = 0; i < collections.Count; i++)
                 collections[i].PrepareForEditorMode();
         }
-
-        public void ClearBadItems()
+#endif
+        public void ValidateCollections()
         {
             for (int i = 0; i < collections.Count; i++)
             {
-                collections[i].ClearBadItems();
+                ScriptableObjectCollection collectionA = collections[i];
+
+                for (int j = 0; j < collections.Count; j++)
+                {
+                    ScriptableObjectCollection collectionB = collections[j];
+
+                    if (i == j)
+                        continue;
+                    
+                    if (collectionA.GUID == collectionB.GUID)
+                    {
+                        collectionA.GenerateNewGUID();
+                        Debug.LogWarning(
+                            $"Found duplicated GUID between {collectionA} and {collectionB}, please run the validation again to make sure this is fixed");
+                    }
+                }
+
+                for (int j = 0; j < collectionA.Items.Count; j++)
```

Comment:
The naming of the iterator variables gets a little icky in this method. `i`, `j` and `k` mean very different things and have very indescriptive names so I could see this being a bit prone to human error.

Might be worth naming them after what they are instead of a single letter in this particular case.

Reply:
Yeah I do think this case is simple enough and since I'm not inlining variables I don't think it's necessary. But yeah, there's definitely cases where naming it properly would be beneficial